code/game/machinery/flasher.dm

```diff
@@ -22,9 +22,14 @@
 	base_state = "pflash"
 	density = 1
 
-/obj/machinery/flasher/New()
+/obj/machinery/flasher/New(loc, ndir = 0, built = 0)
 	..() // ..() is EXTREMELY IMPORTANT, never forget to add it
-	bulb = new /obj/item/device/assembly/flash/handheld(src)
+	if(built)
+		dir = ndir
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -28 : 28)
+		pixel_y = (dir & 3)? (dir ==1 ? -28 : 28) : 0
+	else
+		bulb = new /obj/item/device/assembly/flash/handheld(src)

```

Comment:
why why WHY SHOULD I MKDE THATB EA PROC? I FUCING REFACTORED WALL FRAMES ALREAADY iN #11433 NOT GOIN TO TUCH IT AAGIN